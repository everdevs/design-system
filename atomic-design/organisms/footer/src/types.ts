import { PropsWithTheme } from "@evernest/theme";

export type FooterElement = HTMLDivElement;

export interface StyledFooterProps extends PropsWithTheme<FooterElement> {
	/** Add types */
}

export interface FooterProps extends FooterElement {
	/** Add types */
}
