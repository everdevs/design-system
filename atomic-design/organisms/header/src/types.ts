import { PropsWithTheme } from "@evernest/theme";

export type HeaderElement = HTMLDivElement;

export interface StyledHeaderProps extends PropsWithTheme<HeaderElement> {
	/** Add types */
	translucent?: boolean;
	fadeOffset?: number;
	plain?: boolean;
	fixed?: boolean;
	sticky?: boolean;
}
