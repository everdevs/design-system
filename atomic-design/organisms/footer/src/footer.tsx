import { css } from "@emotion/core";
import styled from "@emotion/styled";
import React from "react";
import { StyledFooterProps, FooterProps } from "./types";

export const StyledFooter = styled.div<StyledFooterProps>`
	padding: var(--spacing-xl) 0;
	${({ theme: { palette } }) => css`
		background: ${palette.darkGrey.css};
		color: ${palette.white.css};
	`};
`;

export const Footer = React.forwardRef<FooterProps, StyledFooterProps>((props, ref) => (
	<StyledFooter {...props} ref={ref} />
));
