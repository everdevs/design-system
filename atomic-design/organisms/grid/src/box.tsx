import { css } from "@emotion/react";
import styled from "@emotion/styled";
import React from "react";
import { GUTTER, PADDING } from "./constants";
import { BoxProps } from "./types";

const StyledBox = styled.div<BoxProps>`
	flex: 1;
	${({ flex }) => css`
		display: ${flex ? "flex" : "block"};
	`};
	${({ removePadding, removeGutter, addGutter, addPadding }) => {
		if (removePadding) {
			if (removeGutter) {
				return css`
					margin-left: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
					margin-right: calc((var(${PADDING}) + var(${GUTTER})) * -1px);
				`;
			}
			return css`
				margin-left: calc(var(${PADDING}) * -1px);
				margin-right: calc(var(${PADDING}) * -1px);
			`;
		}
		if (removeGutter) {
			return css`
				margin-left: calc(var(${GUTTER}) * -1px);
				margin-right: calc(var(${GUTTER}) * -1px);
			`;
		}
		if (addGutter) {
			if (addPadding) {
				return css`
					padding-left: calc(var(${PADDING}) * 1px);
					padding-right: calc(var(${PADDING}) * 1px);
					margin-left: calc(var(${GUTTER}) * 1px);
					margin-right: calc(var(${GUTTER}) * 1px);
				`;
			}
			return css`
				margin-left: calc(var(${GUTTER}) * 1px);
				margin-right: calc(var(${GUTTER}) * 1px);
			`;
		}
		return css`
			padding-left: calc(var(${PADDING}) * 1px);
			padding-right: calc(var(${PADDING}) * 1px);
		`;
	}};
`;

export const Box: React.FC<BoxProps> = props => {
	return <StyledBox {...props} />;
};
