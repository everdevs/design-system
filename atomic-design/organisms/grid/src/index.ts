export * from "./box";
export * from "./column";
export * from "./constants";
export * from "./flex";
export * from "./hidden";
export * from "./grid";
export * from "./grid-overlay";
export * from "./row";
export * from "./stage";
export * from "./types";
