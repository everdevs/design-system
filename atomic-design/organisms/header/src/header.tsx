import styled from "@emotion/styled";
import React from "react";
import { StyledHeaderProps, HeaderProps } from "./types";
import { css } from "@emotion/core";

export const StyledHeader = styled.div<StyledHeaderProps>`
	height: var(--header-height, inherit);
	display: flex;
	flex-direction: column;
	justify-content: center;
	top: 0;
	left: 0;
	right: 0;
	z-index: 100;
	${({ theme: { palette }, fixed, plain, sticky }) => css`
		position: ${fixed ? "fixed" : sticky ? "sticky" : "absolute"};
		background: ${plain ? "inherit" : palette.darkGrey.css};
		color: ${plain ? "currentColor" : palette.white.css};
	`};
`;

export const Header = React.forwardRef<HeaderProps, StyledHeaderProps>((props, ref) => (
	<StyledHeader {...props} ref={ref} />
));
