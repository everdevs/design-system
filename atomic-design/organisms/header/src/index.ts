export * from "./header";
export * from "./types";
export * from "./utils";
