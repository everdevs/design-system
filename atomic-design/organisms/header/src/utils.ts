/**
 * Converts a given value to a float from 0 to 1.
 * The `max` parameter is used to determine the progress while 0 is always the starting point.
 * @param value
 * @param max
 * @example
 *   const opacity = toOpacityValue(5, 10); // 0.5
 * @example
 *   const opacity = toOpacityValue(10, 10); // 1
 * @example
 *   const opacity = toOpacityValue(0, 10); // 0
 */
export const toOpacityValue = (value: number, max: number): number => {
	return value === max || max === 0 ? 1 : value === 0 ? 0 : Math.min(value, max) / max;
};
