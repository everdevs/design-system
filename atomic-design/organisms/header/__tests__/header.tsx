import React from "react";
import { mount } from "enzyme";
import { Header, StyledHeaderProps } from "../src";
import { createTree, withTheme, Wrapper } from "@evernest/dev-helpers";
import { theme } from "@evernest/theme";

const ThemedHeader = withTheme<StyledHeaderProps>(Header, theme);

test("Header renders children", () => {
	const text = "Hello World!";
	const wrapper = mount(<ThemedHeader>{text}</ThemedHeader>);
	expect(wrapper.text()).toEqual(text);
});

test("Header allows setting classname", () => {
	const className = "test";
	const wrapper = mount(<ThemedHeader className={className} />);
	expect(wrapper).toHaveClassName(className);
});

test("Header allows setting fadeOffset", () => {
	const fadeOffset = 100;
	const wrapper = mount(<ThemedHeader translucent fadeOffset={fadeOffset} />);
	expect(wrapper.props().fadeOffset).toEqual(fadeOffset);
});

test("Header can be fixed", () => {
	const tree = createTree(<ThemedHeader fixed />);
	expect(tree).toMatchSnapshot();
});

test("Header can be sticky", () => {
	const tree = createTree(<ThemedHeader sticky />);
	expect(tree).toMatchSnapshot();
});

test("Header can be plain", () => {
	const tree = createTree(<ThemedHeader plain />);
	expect(tree).toMatchSnapshot();
});

test("Header can be translucent", () => {
	const tree = createTree(<ThemedHeader translucent />);
	expect(tree).toMatchSnapshot();
});
