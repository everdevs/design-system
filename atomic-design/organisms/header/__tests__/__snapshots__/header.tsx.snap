// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header can be fixed 1`] = `
<div
  className="css-es0qts"
>
  <div
    className="css-sfg84s"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="css-1fxu3in"
  />
</div>
`;

exports[`Header can be plain 1`] = `
<div
  className="css-rzbc7y"
>
  <div
    className="css-1m3uhup"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="css-1fxu3in"
  />
</div>
`;

exports[`Header can be sticky 1`] = `
<div
  className="css-1y5m4me"
>
  <div
    className="css-sfg84s"
    style={
      Object {
        "opacity": 1,
      }
    }
  />
  <div
    className="css-1fxu3in"
  />
</div>
`;

exports[`Header can be translucent 1`] = `
<div
  className="css-e45j2r"
>
  <div
    className="css-sfg84s"
    style={
      Object {
        "opacity": 0,
      }
    }
  />
  <div
    className="css-1fxu3in"
  />
</div>
`;
